packages/node-type-resolver/src/Reflection/BetterReflection/SourceLocatorProvider/DynamicSourceLocatorProvider.php

```diff
@@ -54,8 +54,8 @@ public function provide(): SourceLocator
             $sourceLocators[] = new OptimizedSingleFileSourceLocator($this->fileNodesFetcher, $file);
         }
 
-        foreach ($this->filesByDirectory as $files) {
-            $sourceLocators[] = new OptimizedDirectorySourceLocator($this->fileNodesFetcher, $files);
+        foreach ($this->filesByDirectory as $file) {
```

Comment:
This case `x by y` needs to be covered.

```php
foreach ($files By Directory as $files) {
    // ...
}
```

Reply:
implemented https://github.com/rectorphp/rector/pull/5696/commits/f589ba3949addb09901e08a7a0e0ed5841c7e52d